src/llama-model.cpp

```diff
@@ -3253,7 +3253,9 @@ bool llama_model::load_tensors(llama_model_loader & ml) {
                     const uint32_t v_dim              = head_dim;
                     const int64_t num_attention_heads = hparams.n_head();
                     const int64_t q_num_heads         = num_attention_heads;
-                    const int64_t dt_dim              = std::max(64, int(hparams.n_embd / 16));
+                    const int64_t dt_dim              = hparams.ssm_dt_rank > 0
```

Comment:
Since this is in the `LLM_ARCH_PLAMO2` case, I think this is not actually getting stimulated in this PR. I'd be curious if you made the same change for the `nemotronh` model what the result would be.